### PR TITLE
Fix/6185: Windows compatibiliy: unspecified error when uploading a forbidden file

### PR DESCRIPTION
**Pull request by samin-z**
* Resolves: # [[6185](https://github.com/nextcloud/deck/issues/6185)]
* Target version: main

### Summary
Fix for showing proper user-friendly error message when uploading a forbidden file name.

### TODO

- [ ] ...

### Checklist

- [x] Code is properly formatted
- [x] Sign-off message is added to all commits
- [x] Tests (unit, integration, api and/or acceptance) are included
- [x] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

File: lib/Service/FilesAppService.php [hunk 1]
```diff
@@ -29,6 +33,12 @@
 use Psr\Log\LoggerInterface;
 
 class FilesAppService implements IAttachmentService, ICustomAttachmentService {
+	/**
+	 * With intensive testing, the only case of sanitization is "/" to ":" before the code runs, it's mapped now for better UX.
```
Comment (juliusknorr) [inline lib/Service/FilesAppService.php]: Can you tell how you triggered this? I would be surprised if we really need such a mapping, but maybe I miss some case here

File: lib/Service/FilesAppService.php [hunk 2]
```diff
@@ -307,4 +321,18 @@ private function getShareForAttachment(Attachment $attachment): IShare {
...
+
+	private function validateFilename(string $fileName): void {
+		try {
+			$this->filenameValidator->validateFilename($fileName);
+		} catch (\Exception $e) {
+			$errorMessage = $e->getMessage();
+			if (preg_match('/\\\\?["\']([^"\']+)\\\\?["\'] is not allowed/', $errorMessage, $matches)) {
+				$invalidCharacter = stripslashes($matches[1]);
+				$invalidCharacter = self::SANITIZED_CHAR_MAPPING[$invalidCharacter] ?? $invalidCharacter;
+				throw new BadRequestException($this->l10n->t('Invalid filename. The character "%s" is not allowed in filenames.', [$invalidCharacter]));
```
[the remark this answers is not in the document]
Comment (samin-z) [reply lib/Service/FilesAppService.php]: i added the regex for the edge case of ":". it was over engineering, i will updated it with simple str_replace.

File: lib/Service/FilesAppService.php (hunk 1 above)
Comment (samin-z) [reply lib/Service/FilesAppService.php]: i uploaded a file "test/file.txt" and in the error it returned "test:file.txt", so i found it as a edge case.

Comment (juliusknorr) [reply lib/Service/FilesAppService.php]: Ah did you test on windows then? I was just thinking about macOS/Linux but there the : makes sense as a replacement then indeed.

Comment (juliusknorr) [reply lib/Service/FilesAppService.php]: Let's keep as you proposed, can you add a comment to the code with a hint that this is specific to mac os with above reasoning? That way it is easier to understand if checking the code later on. Approving already as the other changes look good.

**Comment by juliusknorr** [comment]
The integration test failure does not seem related, let me rebase to see

**Comment by samin-z** [comment]
@juliusknorr, thank you. i ran unittest and integration test locally and both passed with no issue.
